### PR TITLE
New Auth Methods

### PR DESCRIPTION
**Pull request by douxc**
# Checklist:

> [!IMPORTANT]  
> Please review the checklist below before submitting your pull request.

- [x] Please open an issue before creating a PR or link to an existing issue
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I ran `dev/reformat`(backend) and `cd web && npx lint-staged`(frontend) to appease the lint gods

# Description
All auth methods now configurable

## Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update, included: [Dify Document](https://github.com/langgenius/dify-docs)
- [x] Improvement, including but not limited to code refactoring, performance optimization, and UI/UX improvement
- [ ] Dependency upgrade

### CODE REVIEW

File: web/yarn.lock
Comment (crazywoola) [inline web/yarn.lock:1]: This file should not be committed if `package.json` does not have changes

File: web/service/common.ts [hunk 1]
```diff
@@ -156,12 +156,12 @@ export const updatePluginProviderAIKey: Fetcher<UpdateOpenAIKeyResponse, { url:
   return post<UpdateOpenAIKeyResponse>(url, { body })
 }
 
-export const invitationCheck: Fetcher<CommonResponse & { is_valid: boolean; workspace_name: string }, { url: string; params: { workspace_id: string; email: string; token: string } }> = ({ url, params }) => {
-  return get<CommonResponse & { is_valid: boolean; workspace_name: string }>(url, { params })
+export const invitationCheck: Fetcher<CommonResponse & { is_valid: boolean; data: { workspace_name: string;email: string;workspace_id: string } }, { url: string; params: { workspace_id?: string; email?: string; token: string } }> = ({ url, params }) => {
```
Comment (crazywoola) [inline web/service/common.ts]:
```suggestion
export const invitationCheck: Fetcher<CommonResponse & { is_valid: boolean; data: { workspace_name: string; email: string; workspace_id: string } }, { url: string; params: { workspace_id?: string; email?: string; token: string } }> = ({ url, params }) => {
```

File: web/service/base.ts [hunk 2]
```diff
@@ -113,6 +113,7 @@ function unicodeToChar(text: string) {
 }
 
 function requiredWebSSOLogin() {
+  console.log(globalThis.location.pathname)
```
Comment (crazywoola) [inline web/service/base.ts]:
```suggestion
```

File: web/app/signin/oneMoreStep.tsx [hunk 3]
```diff
@@ -46,11 +46,12 @@ const reducer = (state: IState, action: any) => {
 const OneMoreStep = () => {
   const { t } = useTranslation()
   const router = useRouter()
+  const searchParams = useSearchParams()
   // const { locale } = useContext(I18n)
```
Comment (crazywoola) [inline web/app/signin/oneMoreStep.tsx]:
```suggestion
```

File: web/app/signin/oneMoreStep.tsx [hunk 4]
```diff
@@ -3,7 +3,7 @@ import React, { useEffect, useReducer } from 'react'
 import { useTranslation } from 'react-i18next'
 import Link from 'next/link'
 import useSWR from 'swr'
-import { useRouter } from 'next/navigation'
+import { useRouter, useSearchParams } from 'next/navigation'
 // import { useContext } from 'use-context-selector'
```
Comment (crazywoola) [inline web/app/signin/oneMoreStep.tsx]:
```suggestion
```

File: web/app/reset-password/check-code/countdown.tsx [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+    onResend?.()
+  }
+
+  useEffect(() => {
+    localStorage.setItem(COUNT_DOWN_KEY, `${time}`)
+  }, [time])
+
+  return <p className='text-text-tertiary text-xs'>
+    <span>{t('login.checkCode.didNotReceiveCode')}</span>
+    {time > 0 && <span>{Math.round(time / 1000)}s</span>}
```
Comment (crazywoola) [inline web/app/reset-password/check-code/countdown.tsx]: Suggest using `?` operator.

File: web/app/reset-password/page.tsx [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+        Toast.notify({
+          type: 'error',
+          message: t('login.error.emailInValid'),
+        })
+        return
+      }
+      setIsLoading(true)
+      const res = await sendResetPasswordCode(email)
+      if (res.result === 'success') {
+        localStorage.setItem('leftTime', '59000')
```
Comment (crazywoola) [inline web/app/reset-password/page.tsx]: This should reuse the defined const.

File: web/app/signin/components/mail-and-code-auth.tsx [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+        Toast.notify({
+          type: 'error',
+          message: t('login.error.emailInValid'),
+        })
+        return
+      }
+      setIsLoading(true)
+      const ret = await sendEMailLoginCode(email)
+      if (ret.result === 'success') {
+        localStorage.setItem('leftTime', '59000')
```
Comment (crazywoola) [inline web/app/signin/components/mail-and-code-auth.tsx]: This should use the defined const as well.

File: web/app/signin/components/social-auth.tsx [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+}
+
+export default function SocialAuth(props: SocialAuthProps) {
+  const { t } = useTranslation()
+  const searchParams = useSearchParams()
+  const invite_token = decodeURIComponent(searchParams.get('invite_token') || '')
+
+  const getOAuthLink = (href: string) => {
+    const url = getPurifyHref(`${apiPrefix}${href}`)
+    if (invite_token)
```
Comment (crazywoola) [inline web/app/signin/components/social-auth.tsx]:
```
const invite_token = decodeURIComponent(searchParams.get('invite_token') || '')
```

I think should be neither empty nor undefined.

**Review by crazywoola — CHANGES_REQUESTED** [review]
See comments

File: web/app/reset-password/set-password/page.tsx [hunk 9]
```diff
@@ -0,0 +1,158 @@
...
+                  {t('common.account.confirmPassword')}
+                </label>
+                <div className="mt-1 relative rounded-md shadow-sm">
+                  <input
+                    id="confirmPassword"
+                    type='password'
+                    value={confirmPassword}
+                    onChange={e => setConfirmPassword(e.target.value)}
+                    placeholder={t('login.confirmPasswordPlaceholder') || ''}
+                    className={'appearance-none block w-full rounded-lg pl-[14px] px-3 py-2 border border-gray-200 hover:border-gray-300 hover:shadow-sm focus:outline-none focus:ring-primary-500 focus:border-primary-500 placeholder-gray-400 caret-primary-600 sm:text-sm pr-10'}
```
Comment (iamjoel) [inline web/app/reset-password/set-password/page.tsx]: Change `border-gray-200`, `hover:border-gray-300` , `ring-primary-500`, `focus:border-primary-500`, `placeholder-gray-400`, `caret-primary-600` to support dark theme classname.

File: web/app/reset-password/set-password/page.tsx [hunk 10]
```diff
@@ -0,0 +1,158 @@
...
+              </div>
+            </div>
+          </div>
+        </div>
+      )}
+      {showSuccess && (
+        <div className="flex flex-col md:w-[400px]">
+          <div className="w-full mx-auto">
+            <div className="mb-3 flex justify-center items-center w-20 h-20 p-5 rounded-[20px] border border-gray-100 shadow-lg text-[40px] font-bold">
+              <RiCheckboxCircleFill className='w-10 h-10 text-[#039855]' />
```
Comment (iamjoel) [inline web/app/reset-password/set-password/page.tsx]: `text-[#039855]`

File: web/app/reset-password/set-password/page.tsx (hunk 9 above)
Comment (douxc) [reply web/app/reset-password/set-password/page.tsx]: will use Input component instead

File: web/app/signin/components/mail-and-password-auth.tsx [hunk 11]
```diff
@@ -0,0 +1,132 @@
...
+          type={showPassword ? 'text' : 'password'}
+          autoComplete="current-password"
+          placeholder={t('login.passwordPlaceholder') || ''}
+          className='px-3 h-9'
+        />
+        <div className="absolute inset-y-0 right-0 flex items-center pr-3">
+          <button
+            type="button"
+            onClick={() => setShowPassword(!showPassword)}
+            className="text-gray-400 hover:text-gray-500 focus:outline-none focus:text-gray-500"
```
Comment (iamjoel) [inline web/app/signin/components/mail-and-password-auth.tsx]: text-gray-400 hover:text-gray-500 focus:outline-none focus:text-gray-500

File: web/app/signin/components/social-auth.tsx [hunk 12]
```diff
@@ -0,0 +1,62 @@
...
+      return `${url}?${searchParams.toString()}`
+
+    return url
+  }
+  return <>
+    <div className='w-full'>
+      <a href={getOAuthLink('/oauth/login/github')}>
+        <Button
+          disabled={props.disabled}
+          className='w-full hover:!bg-gray-50'
```
Comment (iamjoel) [inline web/app/signin/components/social-auth.tsx]: bg-gray-50

File: web/app/signin/components/sso-auth.tsx [hunk 13]
```diff
@@ -0,0 +1,72 @@
...
+}) => {
+  const router = useRouter()
+  const { t } = useTranslation()
+  const searchParams = useSearchParams()
+  const invite_token = decodeURIComponent(searchParams.get('invite_token') || '')
+
+  const [isLoading, setIsLoading] = useState(false)
+
+  const handleSSOLogin = () => {
+    setIsLoading(true)
```
Comment (iamjoel) [inline web/app/signin/components/sso-auth.tsx:27]: Use enum.

File: web/app/signin/invite-settings/page.tsx [hunk 14]
```diff
@@ -0,0 +1,154 @@
...
+      recheck()
+    }
+  }, [language, name, recheck, setLocaleOnClient, timezone, token, router])
+
+  if (!checkRes)
+    return <Loading />
+  if (!checkRes.is_valid) {
+    return <div className="flex flex-col md:w-[400px]">
+      <div className="w-full mx-auto">
+        <div className="mb-3 flex justify-center items-center w-20 h-20 p-5 rounded-[20px] border border-gray-100 shadow-lg text-[40px] font-bold">🤷‍♂️</div>
```
Comment (iamjoel) [inline web/app/signin/invite-settings/page.tsx]: border-gray-100

File: web/app/signin/normalForm.tsx [hunk 15]
```diff
@@ -1,272 +1,122 @@
...
 
   return (
     <>
-      <div className="w-full mx-auto">
-        <h2 className="text-[32px] font-bold text-gray-900">{t('login.pageTitle')}</h2>
-        <p className='mt-1 text-sm text-gray-600'>{t('login.welcome')}</p>
-      </div>
-
       <div className="w-full mx-auto mt-8">
+        {isInviteLink
```
Comment (iamjoel) [inline web/app/signin/normalForm.tsx:93]: text-gray-900,text-gray-600

File: web/app/signin/normalForm.tsx [hunk 16]
```diff
@@ -1,272 +1,122 @@
...
-            </div>
-          )}
+          <div className="flex flex-col gap-3 mt-6">
+            {systemFeatures.enable_social_oauth_login && <SocialAuth />}
+            {systemFeatures.sso_enforced_for_signin && <div className='w-full'>
+              <SSOAuth protocol={systemFeatures.sso_enforced_for_signin_protocol} />
+            </div>}
+          </div>
 
+          {showORLine && <div className="relative mt-6">
```
Comment (iamjoel) [inline web/app/signin/normalForm.tsx:110]: gray-300

**Review by iamjoel — CHANGES_REQUESTED** [review]

**Review by crazywoola — DISMISSED** [review]

**Review by crazywoola — DISMISSED** [review]

**Review by iamjoel — APPROVED** [review]

**Review by GarfieldDai — APPROVED** [review]
LGTM

**Review by crazywoola — APPROVED** [review]

**Review by 9764319764319764 — COMMENTED** [review]
hamed

**Pull request merged**